src/grpc.ts

```diff
@@ -141,33 +174,81 @@ export class GrpcClient {
 
   /**
    * Load grpc proto services with the specific arguments.
-   * @param {Array=} args - The argument list to be passed to grpc.load().
-   * @return {Object} The gRPC loaded result (the toplevel namespace object).
+   * @param filename The .proto file to load, or the array
+   *   [filename, _, options].
+   * @param _ Unused. Formerly the 'format' parameter for grpc.load.
+   * @param options Options for loading the gRPC file. Fields can either be from
+   * options passed to grpc.load, or grpcProtoLoader.loadSync.
+   * @return The gRPC loaded result (the toplevel namespace object).
    */
-  load(args) {
-    if (!args) {
-      args = [];
-    } else if (!Array.isArray(args)) {
-      args = [args];
+  load(filename: GrpcLoadFileArg, _: null, options: GrpcLoadOptions):
+      grpc.GrpcObject;
+  load(filename: GrpcLoadArgs): grpc.GrpcObject;
+  load(
+      filename: GrpcLoadFileArg|GrpcLoadArgs, _?: null,
+      options?: GrpcLoadOptions): grpc.GrpcObject {
+    if ((Array.isArray as (x) => x is GrpcLoadArgs)(filename)) {
+      options = filename[2] || filename[1] || {};
+      filename = filename[0] as GrpcLoadFileArg;
+    }
+    let root: string|null = null;
```

Comment:
It would be nice to avoid `null` here if we could and just rely on `undefined`.  Does this get assigned?